engine/src/main/java/org/hibernate/search/query/engine/impl/AbstractHSQuery.java

```diff
@@ -405,29 +405,45 @@ private void validateSortField(SortField sortField, DocumentFieldMetadata fieldM
 			}
 	}
 
-	private void validateNumericSortField(SortField sortField, NumericEncodingType numericEncodingType) {
+	private void validateNumericSortField(SortField sortField, NumericEncodingType indexNumericEncodingType) {
+		final NumericEncodingType sortNumericEncodingType;
 		switch ( sortField.getType() ) {
 			case BYTES:
 			case INT:
-				validateNumericEncodingType( sortField, numericEncodingType, NumericEncodingType.INTEGER );
+				sortNumericEncodingType = NumericEncodingType.INTEGER;
 				break;
 			case LONG:
-				validateNumericEncodingType( sortField, numericEncodingType, NumericEncodingType.LONG );
+				sortNumericEncodingType = NumericEncodingType.LONG;
 				break;
 			case DOUBLE:
-				validateNumericEncodingType( sortField, numericEncodingType, NumericEncodingType.DOUBLE );
+				sortNumericEncodingType = NumericEncodingType.DOUBLE;
 				break;
 			case FLOAT:
-				validateNumericEncodingType( sortField, numericEncodingType, NumericEncodingType.FLOAT );
+				sortNumericEncodingType = NumericEncodingType.FLOAT;
 				break;
 			default:
-				throw LOG.sortTypeDoesNotMatchFieldType( String.valueOf( sortField.getType() ), String.valueOf( numericEncodingType ), sortField.getField() );
+				throw LOG.sortTypeDoesNotMatchFieldType( String.valueOf( sortField.getType() ),
+						String.valueOf( indexNumericEncodingType ), sortField.getField() );
 		}
+		if ( NumericEncodingType.UNKNOWN.equals( indexNumericEncodingType ) ) {
+			/*
+			 * The actual encoding type is unknown, so we can't validate more.
+			 * This happens most notably when using custom numeric field bridges.
+			 * We can't ask the user to add more metadata, since there is no annotation for
```

Comment:
So, as we discussed on HipChat, I was simply doing things wrong.
I changed the comments as requested: https://github.com/hibernate/hibernate-search/pull/1138/commits/2cf890974551462ce4f7ddb916dbdd8d9e17641d
@gsmet Okay to merge?